demo_modules/cosines/cosines.js

```diff
@@ -56,9 +56,17 @@ class CosinesSketch {
     
     for (var y = 0; y < this.height; ++y) {
       const yp = 360 * y / this.height;
+      const yloc = yspace * 0.5 + yspace * y + yspace * 0.44 * this.scaledCos(p5.radians(t * 0.0391 + (4.65 * yp)));
```

Comment:
Another (optional) choice is to compute the virtual rect bounds in the space of the circles (which you only do once) and just a simple clip detect

Reply:
I thought about doing rect test, but then I need to calculate Y *AND* X, and I thought it might be, you know, marginally faster if, well, I did it by hand.